File: 14 - Flatten a Nested Object/solution.js
function flattenObject(obj) {
    function helper(obj, path = '', result = {}) {
        for (let key in obj) {
            if (obj.hasOwnProperty(key)) {
                let newKey = path ? `${path}.${key}` : key;
                if (typeof obj[key] === 'object' && obj[key] !== null) {
                    helper(obj[key], newKey, result);
                } else {
                    result[newKey] = obj[key];
                }
            }
        }
        return result;
    }
    return helper(obj);
}



// Test Cases
console.log(flattenObject({ a: { b: { c: 1 }, d: 2 } }));

File: 14 - Flatten a Nested Object/solution.py
def flatten_object(obj):
    def helper(obj, path='', result={}):
        for key in obj:
            new_key = f'{path}.{key}' if path else key

            if isinstance(obj[key], dict) and obj[key] != None:
                helper(obj[key], new_key, result)
            else:
                result[new_key] = obj[key]
        return result
    return helper(obj)


# Test Cases
print(flatten_object({'a': {'b': {'c': 1}, 'd': 2}})) # {'a.b.c': 1, 'a.d': 2}